File: problem/ai-battle-series/05-guess-number/src/main.py
import sys


def main():
    line = sys.stdin.readline().strip()
    if not line:
        return
    n = int(line)
    low, high = 1, n
    while low <= high:
        mid = (low + high) // 2
        print(f"guess {mid}", flush=True)
        resp = sys.stdin.readline().strip()
        if resp == "CORRECT":
            return
        if resp == "HIGHER":
            low = mid + 1
        else:
            high = mid - 1


if __name__ == "__main__":
    main()

File: problem/ai-battle-series/05-guess-number/src/main.c
#include <stdio.h>
#include <string.h>

int main(void) {
    int n = 0;
    if (scanf("%d", &n) != 1) {
        return 0;
    }
    int low = 1;
    int high = n;
    char response[16];

    while (low <= high) {
        int mid = low + (high - low) / 2;
        printf("guess %d\n", mid);
        fflush(stdout);
        if (scanf("%15s", response) != 1) {
            return 0;
        }
        if (strcmp(response, "CORRECT") == 0) {
            return 0;
        }
        if (strcmp(response, "HIGHER") == 0) {
            low = mid + 1;
        } else {
            high = mid - 1;
        }
    }
    return 0;
}
